tree_segment.py

```diff
@@ -0,0 +1,434 @@
+import laspy
+import matplotlib.pyplot as plt
+import math
+import numpy as np
+import random
+import time
+import pandas as pd
+import argparse
+import re
+import os
+
+
+class Point_Cloud:
+    # object that stores the points in a point cloud, initialized from a las file
+    # points can be enabled or disabled, disabled points are ignored when searching for stems
+    def __init__(self, file_name, verbose=False):
+        all_data = laspy.read(file_name)
+        height_of_points_above_ground = np.array(
+            [point[-1] for point in all_data.points.array])
+        self.points = [Point(xyz, height) for xyz, height in zip(
+            all_data.xyz, height_of_points_above_ground)]
+        # internal, to access this use enabled_points()
+        self.__enabled_points__ = self.points
+        self.verbose = verbose
+
+    def enabled_points(self):
+        # filters the set of points for just those enabled, then returns that set of enabled points
+        self.__enabled_points__ = filter_for_enabled(self.__enabled_points__)
+        return self.__enabled_points__
+
+    def find_highest_point(self):
+        # returns the highest point in the point cloud, ie the one with the greatest z coordinate
+        return max(self.enabled_points(), key=lambda point: point.z())
+
+    def find_stem(self, min_height=.5, fail_to_find_disable_radius=.2, descendents_height=.3):
```

Comment:
This function updates the class state by changing the set of enabled points. Please document this in the function comment, or restructure the algorithm to separate the parts that affect the state.

https://realpython.com/python-logging/
> For example, a function that mutates one of its arguments as a side effect should note that in its docstring.

Meta Commentary:
State manipulation is often tricky to reason about and deserves extra caution. That said, I think you're doing a pretty great job already:
* Only one bit of state per point.
* The state evolves monotonically over time.
* Most/all mutation goes through a small number of functions (e.g. `disable_stem_region`).

Reply:
Updated docstring to reflect this.